### PR TITLE
Feat/guide/working with data

### PR DESCRIPTION
**Pull request by pdaoust**
Add an overview page for working with data in Holochain. Covers actions/records, entries, links, CRUD, and source chains.

### CODE REVIEW

File: src/pages/build/working-with-data.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+## Entries, actions, and records: primary data
+
+Data in Holochain takes the shape of a **record**. This record can store lots of different kinds of things, but the most important thing common to all records is the [**action**](https://docs.rs/holochain_zome_types/latest/holochain_zome_types/action/enum.Action.html): one participant's intent to manipulate their own state or the application's shared database state. All actions contain:
+
+* The **agent ID** of the author
+* A timestamp
+* The type of action
+* The hash of the previous action in the author's history of state changes, called their **source chain** (note: the first action in their chain doesn't contain this)
+* The index of the action in the author's source chain
+* The **weight** of the action, which is a calculation of the cost of storing the action and can be used for rate limiting
```
Comment (c12i) [inline src/pages/build/working-with-data.md:18]: I think adding a link to the docs for each method call they can use to get the respective action data would be very helpful

**Review by c12i — COMMENTED** [review]
This documentation looks great, and the illustrations effectively help with visualization! However, I believe adding some code examples would significantly enhance its utility for hApp developers. It would be beneficial to include examples demonstrating how records/ links are created, updated, deleted, and read using methods like `create_entry`, `get`, `get_details`, `delete_link`, `create_link` etc.

What do you think?

**Comment by pdaoust** [comment]
@c12i

> However, I believe adding some code examples would significantly enhance its utility for hApp developers.

Yes, absolutely. This page is just for the overview, then other pages like [Entries](https://developer.holochain.org/build/entries/) will have examples of using those functions. I ought to link more to them from this page though.

**Comment by c12i** [comment]
> I ought to link more to them from this page though.

Fair enough, the links should suffice then

**Review by c12i — APPROVED** [review]

**Review by c12i — APPROVED** [review]

**Pull request merged**